cmd/scollector/collectors/nexpose.go

```diff
@@ -0,0 +1,405 @@
+			}
+			Add(&md, "nexpose.asset.risk_score", asset.RiskScore, assetTags, metadata.Gauge, metadata.Score, descRiskScore)
+			Add(&md, "nexpose.asset.vuln_count", asset.VulnCount, assetTags, metadata.Gauge, metadata.Vulnerabilities, descVulnCount)
+			Add(&md, "nexpose.asset.exploit_count", asset.ExploitCount, assetTags, metadata.Gauge, metadata.Vulnerabilities, descExploitCount)
+			Add(&md, "nexpose.asset.malware_count", asset.MalwareCount, assetTags, metadata.Gauge, metadata.Vulnerabilities, descMalwareCount)
+			Add(&md, "nexpose.asset.last_scan", last_scan, assetTags, metadata.Gauge, metadata.Second, descAssetLastScan)
+
+			site.VulnCount += asset.VulnCount
+			site.ExploitCount += asset.ExploitCount
+			site.MalwareCount += asset.MalwareCount
+		}
+
+		Add(&md, "nexpose.site.vuln_count", site.VulnCount, tags, metadata.Gauge, metadata.Vulnerabilities, descVulnCount)
+		Add(&md, "nexpose.site.exploit_count", site.ExploitCount, tags, metadata.Gauge, metadata.Vulnerabilities, descExploitCount)
+		Add(&md, "nexpose.site.malware_count", site.MalwareCount, tags, metadata.Gauge, metadata.Vulnerabilities, descMalwareCount)
+	}
+
+	activeScans, err := c.scanActivity()
+	for _, scan := range *activeScans {
+		if scan.Status == "running" {
+			Add(&md, "nexpose.scan.running", 1, opentsdb.TagSet{"site": siteNames[scan.SiteID]}, metadata.Gauge, metadata.Bool, descScanRunning)
+		}
+	}
+
+	return md, nil
+}
+
+const (
+	descScanRunning   = "Nexpose scan running."
+	descRiskScore     = "Risk score for a given site/device."
+	descVulnCount     = "Number of known vulnerabilities."
+	descExploitCount  = "Number of vulnerabilities exploitable via Metasploit. Subset of vuln_count."
+	descMalwareCount  = "Number of vulnerabilities susceptible to malware attacks. Subset of vuln_count."
+	descAssetLastScan = "How many seconds ago this asset was last scanned."
+)
+
+func (c *nexposeConnection) jsonRequest(method string, location string, v *url.Values) (*[]byte, error) {
+	req, _ := http.NewRequest(method, "https://"+c.Host+location, strings.NewReader(v.Encode()))
+
+	// Some methods require the cookie, some require the header, so we set both.
+	req.AddCookie(c.Cookie)
+	req.Header.Add("nexposeCCSessionID", c.SessionID)
+	req.Header.Add("Content-Type", "application/x-www-form-urlencoded")
+
+	resp, err := c.Client.Do(req)
+	if err != nil {
+		return nil, err
+	}
+	defer resp.Body.Close()
+
+	body, _ := ioutil.ReadAll(resp.Body)
+
+	if err != nil {
+		return nil, err
+	}
+
+	return &body, nil
+}
+
+type jsonTable struct {
+	RowsPerPage  int
+	RecordOffset int
+	TotalRecords int
+	Records      []json.RawMessage
+}
+
+func (t *jsonTable) concat(jsonData *[]byte) error {
+	var newTable jsonTable
+	err := json.Unmarshal(*jsonData, &newTable)
+	if err != nil {
+		return err
+	}
+	t.Records = append(t.Records, newTable.Records...)
+
+	return nil
+}
+
+func (c *nexposeConnection) getJsonTable(location string, params map[string]string) (*jsonTable, error) {
```

Comment:
these functions are a bit unorganized. I would like to see all methods on a type somewhat grouped together.